pyfar/signals/files.py

```diff
@@ -389,15 +390,20 @@ def head_related_impulse_responses(
     tolerance_rad = 0.1 / 180 * np.pi
 
     # get indices of source positions
-    if position == "horizontal":
+    if isinstance(position, str) and position == "horizontal":
         idx = sources.elevation == 0
-    elif position == "median":
+    elif isinstance(position, str) and position == "median":
         idx = sources.lateral == 0
         idx = np.where(idx)[0]
         # sort positions according to polar angle
         polar = sources.polar[idx].flatten()
         idx = (idx[np.argsort(polar)], )
     else:
+
+        # get coordinates - required for more verbose error handling below
+        if isinstance(position, pf.Coordinates):
```

Comment:
maybe convert non coords into pyfar coordintes, we are using them later anyway. In this was its very inefficient

Reply:
and the part below would be more clean

Reply:
Done as suggested